Packs/Linkshadow/Integrations/Linkshadow/Linkshadow.py

```diff
@@ -0,0 +1,253 @@
+            url_suffix='/api/plugin/',
+            data=request_params
+        )
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client, first_fetch_time, apiKey, api_username, plugin_id, action, time_frame):
+    try:
+        alerts = client.fetch_anomaly(apiKey=apiKey, start_time=first_fetch_time,
+                                      api_username=api_username, plugin_id=plugin_id, action=action, time_frame=time_frame)
+        if 'error' in str(alerts.get('message')) or 'success' in str(alerts.get('message')):
+            return "ok"
+        else:
+            return alerts.get('message')
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            return e
+    # return 'ok'
+
+
+def format_JSON_for_fetch_incidents(ls_anomaly):
+
+    anomaly_info = {}
+    anomaly = ls_anomaly
+    anomaly_info['time_seen'] = anomaly['time_seen']if 'time_seen' in anomaly else 'no ls_anomaly time_seen'
+    anomaly_info['category'] = anomaly['category']if 'category' in anomaly else 'no ls_anomaly category'
+    anomaly_info['anomaly_type'] = anomaly['anomaly_type'] if 'anomaly_type' in anomaly else 'no ls_anomaly anomaly_type'
+    anomaly_info['sip'] = anomaly['sip']if 'sip' in anomaly else 'no ls_anomaly sip'
+    anomaly_info['anomaly_id'] = anomaly['anomaly_id']if 'anomaly_id' in anomaly else 'no ls_anomaly anomaly_id'
+    anomaly_info['inserted_time'] = anomaly['inserted_time']if 'inserted_time' in anomaly else 'no ls_anomaly inserted_time'
+    anomaly_info['smac'] = anomaly['smac']if 'smac' in anomaly else 'no ls_anomaly smac'
+    anomaly_info['bandwidth'] = anomaly['bandwidth']if 'bandwidth' in anomaly else 'no ls_anomaly bandwidth'
+    anomaly_info['score'] = anomaly['score']if 'score' in anomaly else 'no ls_anomaly score'
+    anomaly_info['dport'] = anomaly['dport']if 'dport' in anomaly else 'no ls_anomaly dport'
+    anomaly_info['dmac'] = anomaly['dmac']if 'dmac' in anomaly else 'no ls_anomaly dmac'
+    anomaly_info['sport'] = anomaly['sport']if 'sport' in anomaly else 'no ls_anomaly sport'
+    anomaly_info['dip'] = anomaly['dip']if 'dip' in anomaly else 'no ls_anomaly dip'
+    anomaly_info['desc'] = anomaly['desc']if 'desc' in anomaly else 'no ls_anomaly desc'
+    return anomaly_info
+
+
+def fetch_incidents(client, max_alerts, last_run, first_fetch_time, apiKey, api_username, plugin_id, action, time_frame):
+
+    last_fetch = last_run.get('last_fetch', None)
+    if last_fetch is None:
+        last_fetch = first_fetch_time
+    else:
+        last_fetch = int(last_fetch)
+    latest_created_time = int(last_fetch)
+    incidents = []
+    alerts = client.fetch_anomaly(apiKey=apiKey, start_time=first_fetch_time,
+                                  api_username=api_username, plugin_id=plugin_id, action=action, time_frame=time_frame)
+    alert = {}
+    for alert in alerts:
+
+        if alert == 'data':

```

Comment:
This for loop is redundant.

Reply:
When we call Linkshadow API we will get a List of Anomalies as response . Then we are looping the response and create each anomaly as an Incident in XSOAR (also we are mapping Anomaly fields to XSOAR custom fields)

Reply:
Yeah, I meant that only the first loop is redundant.
```suggestion
    for alert in alerts.get('data'):
```